pkg/ui/react-app/src/thanos/pages/stores/StorePoolPanel.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { FC } from 'react';
+import { Container, Collapse, Table, Badge } from 'reactstrap';
+import { now } from 'moment';
+import { ToggleMoreLess } from '../../../components/ToggleMoreLess';
+import { useLocalStorage } from '../../../hooks/useLocalStorage';
+import { getColor } from '../../../pages/targets/target';
+import { formatRelative, formatTime } from '../../../utils';
+import { Store } from './store';
+import StoreLabels from './StoreLabels';
+
+export type StorePoolPanelProps = { title: string; storePool: Store[] };
+
+export const columns = [
+  'Endpoint',
+  'Status',
+  'Announced LabelSets',
+  'Min Time',
+  'Max Time',
+  'Last Successful Health Check',
+  'Last Message',
+];
+
+export const StorePoolPanel: FC<StorePoolPanelProps> = ({ title, storePool }) => {
+  const [{ expanded }, setOptions] = useLocalStorage(`store-pool-${title}-expanded`, { expanded: true });
+
+  return (
+    <Container fluid>
+      <ToggleMoreLess event={(): void => setOptions({ expanded: !expanded })} showMore={expanded}>
+        <span style={{ textTransform: 'capitalize' }}>{title}</span>
+      </ToggleMoreLess>
+      <Collapse isOpen={expanded}>
+        <Table size="sm" bordered hover>
+          <thead>
+            <tr key="header">
+              {columns.map(column => (
+                <th key={column}>{column}</th>
+              ))}
+            </tr>
+          </thead>
+          <tbody>
+            {storePool.map((store: Store) => {
+              const { name, minTime, maxTime, labelSets, lastCheck, lastError } = store;
+              const health = lastError ? 'down' : 'up';
+              const color = getColor(health);
+
+              return (
+                <tr key={name}>
+                  <td>{name}</td>
+                  <td>
+                    <Badge color={color}>{health.toUpperCase()}</Badge>
+                  </td>
+                  <td>
+                    <StoreLabels labelSet={labelSets} />
+                  </td>
+                  <td>{formatTime(minTime)}</td>
```

Comment:
What about very huge values here? Should we print the infinity sign here if it is huge?

Reply:
I was checking the values and I am getting 
```json
{
    "maxTime": 9223372036854775807,
    "minTime": -62167219200000,
}
```
in the response. The `minTime` is negative and `maxTime` is a very large value, so I thought I need to look deeper, and see from where I am getting these values.

Reply:
Okay I think I found the source.
```go
var (
	minTime = time.Unix(math.MinInt64/1000+62135596801, 0)
	maxTime = time.Unix(math.MaxInt64/1000-62135596801, 999999999)
)
```
These are kinda constant values so I guess we can just check for these values and put infinity sign there.